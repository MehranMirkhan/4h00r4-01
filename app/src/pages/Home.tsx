import React, { useEffect } from "react";
import { IonIcon, IonButton } from "@ionic/react";
import { home, trophy, logIn, settings } from "ionicons/icons";
import { useTranslation } from "react-i18next";
import { useDispatch, useSelector } from "react-redux";

import Page from "src/widgets/Page";
import Slide from "src/widgets/Slide";
import MultiCol from "src/widgets/MultiCol";
import { LinkButton } from "src/widgets/Buttons";

import api from "src/api";
import useAsync from "src/tools/useAsync";
import { logout } from "src/reducers/auth.reducer";
import { setCurrentLevel } from "src/reducers/level.reducer";
import config from "src/app.config.json";

import "./Home.css";

export default function() {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const state = useSelector((state: State) => state);

  // News
  const { call, response } = useAsync();
  useEffect(() => {
    call(api.misc.getActiveNews);
  }, [call]);

  return (
    <Page title={t("Welcome to Puzzles")} showBack={false}>
      {!response ? null : (
        <Slide
          images={response.data.map(
            (x: any) => `${config.base_url}/storage/${x.image}`
          )}
        />
      )}

      <MainButton label={t("Level")} to="/level_list" />
      <MainButton label={t("Daily")} to="/question_list?type=daily" />
      <MainButton label={t("Weekly")} to="/question_list?type=weekly" />

      <MultiCol
        cols={[
          <FooterButton icon={home} to="/home" />,
          <FooterButton icon={trophy} to="/leaderboard" />,
          <FooterButton icon={logIn} to="/auth" />,
          <FooterButton icon={settings} to="/settings" />
        ]}
      />

      <div style={{ borderTop: "1px solid black", paddingTop: 8 }}>
        For Dev:
      </div>
      <IonButton onClick={() => dispatch(logout())}>Log out</IonButton>
      <IonButton onClick={() => dispatch(setCurrentLevel(1))}>
        Reset Level
      </IonButton>
      <IonButton
        onClick={() => {
          dispatch({ type: "RESET" });
          window.location.reload();
        }}
      >
        Reset All
      </IonButton>
      <p>{JSON.stringify(state)}</p>
    </Page>
  );
}

const MainButton = ({ label, to }: any) => (
  <LinkButton to={to} className="main-button" expand="block" size="large">
    {label}
  </LinkButton>
);

const FooterButton = ({ icon, to }: any) => (
  <LinkButton to={to} size="large">
    <IonIcon icon={icon} />
  </LinkButton>
);
