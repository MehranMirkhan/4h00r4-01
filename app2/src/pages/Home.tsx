import React, { useEffect } from "react";
import { connect } from "react-redux";
import { home, trophy, logIn, settings } from "ionicons/icons";

import Page from "src/widgets/Page";
import ImageSlide from "src/widgets/ImageSlide";
import Button from "src/widgets/Button";
import MultiCol from "src/widgets/MultiCol";
import { newsSelector, fetchNewsRequest } from "src/state/news";
import { AppState } from "src/state";

// import Dev from "src/components/Dev";

import "src/i18n";
import config from "src/app.config.json";

export function Home({ news, fetchNews }: IHome) {
  useEffect(() => {
    if (!!fetchNews) fetchNews();
  }, []);
  return (
    <Page title="Welcome to Riddles" showBack={false}>
      {!!news && (
        <ImageSlide
          images={news.map(n => `${config.base_url}/storage/${n.image}`)}
        />
      )}

      <Button text="Level" fluid link href="/level_list" />
      <Button text="Daily" fluid link href="/daily" />
      <Button text="Weekly" fluid link href="/weekly" />

      <MultiCol
        cols={[
          <Button icon={home} link href="/home" />,
          <Button icon={trophy} link href="/leaderboard" />,
          <Button icon={logIn} link href="/auth" />,
          <Button icon={settings} link href="/settings" />
        ]}
      />

      {/* <Dev /> */}
    </Page>
  );
}

interface IHome {
  news?: News[];
  fetchNews?: () => void;
}

const props = (state: AppState) => ({ news: newsSelector(state) });
const actions = { fetchNews: fetchNewsRequest };

export default connect(props, actions)(Home);
