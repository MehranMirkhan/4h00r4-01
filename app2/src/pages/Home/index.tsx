import React from "react";

import Page from "src/widgets/Page";

export default function() {
  return (
    <Page title="Welcome to Riddles" showBack={false}>
      <div>Hello, World!</div>
    </Page>
  );
}
