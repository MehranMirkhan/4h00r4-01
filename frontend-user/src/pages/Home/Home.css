.welcome-card img {
  max-height: 35vh;
  overflow: hidden;
}


.slider {
  padding-bottom: 36px;
  background-color: #444;
  --bullet-background-active: yellow;
}

.slider img {
  width: 100%;
}

.main-button {
  padding: 0 10px;
  margin: 10px 0;
}
