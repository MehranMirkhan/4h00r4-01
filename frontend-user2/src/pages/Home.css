.main-button {
  padding: 0 10px;
  margin: 10px 0;
}
