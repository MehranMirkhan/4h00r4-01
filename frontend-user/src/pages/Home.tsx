import React, { useEffect } from "react";
import { IonIcon, IonButton } from "@ionic/react";
import { settings, logIn } from "ionicons/icons";
import { useTranslation } from "react-i18next";
import { useDispatch, useSelector } from "react-redux";

import Page from "src/widgets/Page";
import Slide from "src/widgets/Slide";
import MultiCol from "src/widgets/MultiCol";
import { LinkButton } from "src/widgets/Buttons";

import api from "src/api";
import useAsync from "src/tools/useAsync";
import { logout } from "src/reducers/auth.reducer";
import config from "src/app.config.json";

import "./Home.css";

export default function() {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const state = useSelector((state: State) => state);

  // News
  const { call, response } = useAsync();
  useEffect(() => {
    call(api.misc.getActiveNews);
  }, [call]);

  return (
    <Page title={t("Welcome to Puzzles")} showBack={false}>
      {!response ? null : (
        <Slide
          images={response.data.map(
            (x: any) => `${config.base_url}/storage/${x.image}`
          )}
        />
      )}

      <MainButton label={t("Level")} to="/level_list" />
      <MainButton label={t("Daily")} to="/question_list?type=daily" />
      <MainButton label={t("Weekly")} to="/question_list?type=weekly" />

      <MultiCol
        cols={[
          <FooterButton icon={settings} to="/settings" />,
          <FooterButton icon={logIn} to="/auth" />,
          <FooterButton icon={settings} to="/settings" />,
          <FooterButton icon={logIn} to="/auth" />
        ]}
      />

      <hr/>
      <span>For Dev:</span>
      <IonButton onClick={() => dispatch(logout())}>Log out</IonButton>
      <p>{JSON.stringify(state)}</p>
    </Page>
  );
}

const MainButton = ({ label, to }: any) => (
  <LinkButton to={to} className="main-button" expand="block" size="large">
    {label}
  </LinkButton>
);

const FooterButton = ({ icon, to }: any) => (
  <LinkButton to={to} size="large">
    <IonIcon icon={icon} />
  </LinkButton>
);
