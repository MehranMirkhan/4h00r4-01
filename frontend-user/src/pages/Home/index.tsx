import {
  IonButtons, IonGrid, IonRow, IonCol,
  IonContent, IonHeader, IonMenuButton,
  IonTitle, IonToolbar, IonPage, IonSlides,
  IonSlide, IonButton, IonIcon, IonItem
} from '@ionic/react';
import { settings, logIn, apps } from 'ionicons/icons';
import React from 'react';

import './Home.css';
import { Translate } from 'react-localize-redux';


const HomePage: React.FC = () => {
  return (
    <IonPage>
      <IonHeader>
        <IonToolbar color="dark">
          <IonButtons slot="start">
            <IonMenuButton />
          </IonButtons>
          <IonTitle><Translate id="pages.home.title" /></IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent>
        <HeroSlide />
        <MainButtons />
        <FooterButtons />
      </IonContent>
    </IonPage>
  );
};

const HeroSlide: React.FC = () => {
  const options = {
    initialSlide: 0,
    speed: 400,
    pager: true,
    paginationClickable: true,
    loop: true,
    spaceBetween: 20,
  };
  return (
    <IonSlides pager options={options} className="slider">
      <IonSlide>
        <img src="/assets/shapes.svg" alt="" />
      </IonSlide>
      <IonSlide>
        <img src="/assets/shapes.svg" alt="" />
      </IonSlide>
      <IonSlide>
        <img src="/assets/shapes.svg" alt="" />
      </IonSlide>
    </IonSlides>
  );
}

const MainButtons: React.FC = () => {
  const btns = [
    { url: "/level", text: "pages.home.level" },
    { url: "/daily", text: "pages.home.daily" },
    { url: "/weekly", text: "pages.home.weekly" },
  ];
  return <>
    {btns.map((btn, i) =>
      <IonButton key={i} className="main-button"
        expand="block" size="large"
        routerLink={btn.url} routerDirection="forward"
      >
        <Translate id={btn.text} />
      </IonButton>
    )}
  </>;
}

const FooterButtons: React.FC = () => {
  const btns = [
    { url: "/settings", icon: settings },
    { url: "/auth", icon: logIn },
    { url: "/settings", icon: settings },
    { url: "/auth", icon: logIn },
  ];
  return (
    <>
      <IonGrid>
        <IonRow style={{ textAlign: "center" }}>
          {btns.map((btn, i) =>
            <IonCol key={i}>
              <IonButton routerLink={btn.url} routerDirection="forward" size="large">
                <IonIcon icon={btn.icon} />
              </IonButton>
            </IonCol>
          )}
        </IonRow>
      </IonGrid>
    </>
  );
}

export default HomePage;
