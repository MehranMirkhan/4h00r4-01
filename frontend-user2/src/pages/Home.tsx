import React from "react";
import { IonIcon } from "@ionic/react";
import { settings, logIn } from "ionicons/icons";
import { useTranslation } from "react-i18next";

import Page from "src/widgets/Page";
import Slide from "src/widgets/Slide";
import MultiCol from "src/widgets/MultiCol";
import LinkButton from "src/widgets/LinkButton";

import "./Home.css";

export default function() {
  const { t } = useTranslation();
  return (
    <Page title={t("Welcome to Puzzles")} showBack={false}>
      <Slide
        images={[
          "/assets/shapes.svg",
          "/assets/shapes.svg",
          "/assets/shapes.svg"
        ]}
      />

      <MainButton label={t("Level")} to="/level_list" />
      <MainButton label={t("Daily")} to="/question_list?type=daily" />
      <MainButton label={t("Weekly")} to="/question_list?type=weekly" />

      <MultiCol
        cols={[
          <FooterButton icon={settings} to="/settings" />,
          <FooterButton icon={logIn} to="/auth" />,
          <FooterButton icon={settings} to="/settings" />,
          <FooterButton icon={logIn} to="/auth" />
        ]}
      />
    </Page>
  );
}

const MainButton = ({ label, to }: any) => (
  <LinkButton to={to} className="main-button" expand="block" size="large">
    {label}
  </LinkButton>
);

const FooterButton = ({ icon, to }: any) => (
  <LinkButton to={to} size="large">
    <IonIcon icon={icon} />
  </LinkButton>
);
