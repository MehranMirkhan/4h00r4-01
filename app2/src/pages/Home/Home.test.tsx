import React from "react";
import { shallow } from "enzyme";

import Home from ".";

describe("<Home />", () => {
  let home: any;
  beforeEach(() => {
    home = shallow(<Home />);
  });
  it("should render", () => {
    expect(home).toBeDefined();
  });
  it("should match snapshot", () => {
    expect(home).toMatchSnapshot();
  });
});
