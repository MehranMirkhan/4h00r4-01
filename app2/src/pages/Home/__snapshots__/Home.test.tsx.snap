// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<_default
  showBack={false}
  title="Welcome to Riddles"
>
  <div>
    Hello, World!
  </div>
</_default>
`;
